### PR TITLE
Add new metrics

### PR DESCRIPTION
**Pull request by yngvar-antonsson**
Adds new metrics, introduced in Tarantool 3.0:

https://github.com/tarantool/tarantool/pull/9059
https://github.com/tarantool/tarantool/issues/8501

I didn't forget about

- [x] Changelog
- [x] Documentation (README and rst)

### CODE REVIEW

File: metrics/tarantool/memtx.lua [hunk 1]
```diff
@@ -120,6 +120,36 @@ local function update()
...
+        utils.set_gauge('memtx_tuples_data_read_view',
+            'Memory held for read views',
+            memtx_stat_3.data.read_view, {kind = "read_view"}, nil, {default = true})
+    collectors_list.memtx_tuples_data_garbage =
+        utils.set_gauge('memtx_tuples_data_garbage',
+            'Memory that is unused and scheduled to be freed',
+            memtx_stat_3.data.garbage, nil, nil, {default = true})
+
+
+    collectors_list.memtx_index_extents_total =
```
Comment (DifferentialOrange) [inline metrics/tarantool/memtx.lua:144]: What's the idea behind "extents" part of the name?

**Review by DifferentialOrange — COMMENTED** [review]
It would be nice to have some basic tests (metrics is set for a proper version and non-zero is there is some data).

File: metrics/tarantool/memtx.lua (hunk 1 above)
Comment (yngvar-antonsson) [reply metrics/tarantool/memtx.lua:144]: I took descriptions from this issue:
https://github.com/tarantool/doc/issues/3422

Comment (DifferentialOrange) [reply metrics/tarantool/memtx.lua:144]: I think just `memtx_index_total` is more descriptive and easy to understand.

File: metrics/tarantool/memtx.lua [hunk 2]
```diff
@@ -122,6 +122,36 @@ local function update()
             'Number of retained `read_view` tuples / number of stories',
             memtx_stat.mvcc.tuples.read_view.retained.total, {kind = "total"}, nil, {default = true})
 
+    -- Tarantool 3.0 memory statistics
+
+    local ok, memtx_stat_3 = pcall(box.stat.memtx)
```
[the remark this answers is not in the document]
Comment (yngvar-antonsson) [reply metrics/tarantool/memtx.lua:127]: In tarantool 2x `box.stat.memtx` is a table, in Tarantool 3x `box.stat.memtx` is a function/table

**Review by DifferentialOrange — APPROVED** [review]

**Pull request merged**